extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -72,8 +69,7 @@ export const SAFEFRAME_ORIGIN = 'https://tpc.googlesyndication.com';
  */
 export function safeframeListener(event) {
   const data = tryParseJson(getData(event));
-  /** Only process messages that are valid Safeframe messages */
-  if (event.origin != SAFEFRAME_ORIGIN || !data) {
```

Comment:
If we're not doing an explicit origin check, then we need to check that the `source` matches the iframe. As is, _any_ iframe on the page (amp-foo, or even subframes inside the safeframe) can now send messages that will be interpreted as if they're from the safeframe directly.

Reply:
That seems like a problem today, since any JS that's executing can, by making a child safeframe, cause a message to be sent from `https://tpc.googlesyndication.com`

Reply:
All those other postmessages would not be the same schema as safeframeListener is expecting, and would be discarded, right? (eg, they don't include a data.p field)? Only messages that are made to be parsed by this listener would actually be processed.

Reply:
The format can be faked easily. The fact that this is currently possible just means it's always had a bug, but now it's turned into an actual security issue.

Reply:
Added `source` check.